### PR TITLE
`Assessment`: Display warning when using grading criterion without feedback

### PR DESCRIPTION
**Pull request by mallgaier**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
The additional Feedback component requires textual feedback to submit the assessment. While a visual warning is shown to the TA when the textual feedback is missing, when using a grading criterion without feedback the warning is currently missing. Because of this, the TA cannot submit the assessment, but also has no indication about the reason for this.

### Description
The additional Feedback component now also displays a visual warning when the TA used a Grading Criterion without Feedback. 

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Exercise with 1 Grading Criterion without Feedback to the students

1. Participate and submit the exercise as student
2. Assess the exercise
3. Drag & Drop the Grading Criterion without Feedback into the additional feedback
4. Make sure the red warning is shown
5. (Add some textual description and submit the assessment)

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
https://user-images.githubusercontent.com/94070506/161821774-d0fd099b-2614-4850-95f9-b1869861beba.mp4

### CODE REVIEW

File: src/main/webapp/app/assessment/assessment-detail/assessment-detail.component.html [hunk 1]
```diff
@@ -47,11 +47,11 @@
                     [readOnly]="readOnly"
                     [disabled]="readOnly"
                     [placeholder]="
-                        assessment.gradingInstruction!
+                        assessment.gradingInstruction! && assessment.gradingInstruction!.feedback!
```
Comment (bassner) [inline src/main/webapp/app/assessment/assessment-detail/assessment-detail.component.html:50]:
```suggestion
                        assessment.gradingInstruction?.feedback
```

File: src/main/webapp/app/assessment/assessment-detail/assessment-detail.component.html [hunk 2]
```diff
@@ -47,11 +47,11 @@
...
                             ? ('artemisApp.assessment.additionalFeedbackCommentPlaceholder' | artemisTranslate)
                             : ('artemisApp.assessment.feedbackCommentPlaceholder' | artemisTranslate)
                     "
-                    [required]="!assessment.gradingInstruction"
+                    [required]="!assessment.gradingInstruction || !assessment.gradingInstruction!.feedback"
```
Comment (bassner) [inline src/main/webapp/app/assessment/assessment-detail/assessment-detail.component.html:54]:
```suggestion
                    [required]="!assessment.gradingInstruction?.feedback"
```

**Review by bassner — CHANGES_REQUESTED** [review]
The usage of the non-null assertion operator ``!`` is wrong here in my opinion (and it was before your changes, too). I'd suggest to change it according to my inline comments! :)

Other occurrences in this file, mostly of ``assessment.gradingInstruction!`` in ``*ngIf``, could and should be changed to omit the ``!``, too, but it's up to you if you want to include it in this PR or not. :)

**Review by bassner — APPROVED** [review]
Thanks for adding the changes. LGTM 👍

**Review by maximiliansoelch — APPROVED** [review]
Tested on TS3, works as described.
Code changes look good 👍

**Pull request merged**